pynamodb/indexes.py

```diff
@@ -48,16 +48,25 @@ class Index(Generic[_M], metaclass=IndexMeta):
     Base class for secondary indexes
     """
     Meta: Any = None
+    _model: _M
 
     def __init__(self) -> None:
         if self.Meta is None:
             raise ValueError("Indexes require a Meta class for settings")
         if not hasattr(self.Meta, "projection"):
             raise ValueError("No projection defined, define a projection for this class")
 
-    @classmethod
+    class classorinstancemethod:
```

Comment:
nit - if we're keeping it in this file, I'd move it to module scope and prefix with underscore. But maybe move it to pynamodb/_compat.py?